src/server/pfs/server/trigger.go

```diff
@@ -131,16 +131,15 @@ func (d *driver) isTriggered(txnCtx *txncontext.TransactionContext, t *pfs.Trigg
 		ci := newHead
 		var commits int64
 		for commits < t.Commits {
-			commits++
-			if ci.ParentCommit != nil && oldHead.Commit.Id != ci.ParentCommit.Id {
-				var err error
-				ci, err = d.resolveCommit(txnCtx.SqlTx, ci.ParentCommit)
-				if err != nil {
-					return false, err
-				}
-			} else {
+			if oldHead.Commit.Id == ci.Commit.Id || ci.ParentCommit == nil {
 				break
 			}
+			var err error
+			ci, err = d.resolveCommit(txnCtx.SqlTx, ci.ParentCommit)
+			if err != nil {
+				return false, err
+			}
+			commits++
```

Comment:
By moving the count to the end, the code will break before it counts the initial commit (because the initial commit's id == ci.commit.ID)

Reply:
That's not a problem. The loop invariant is that `newHead != oldHead`. We terminate the loop early if that is no longer true, so we don't want to increment the count before that.